cuberite/cuberite
# Leads, leashed mobs and leash knots on fences

- Adds leads handling, leashed mobs and leash knots on fences.
- Drops appropriate pickup on unleash.
- Save and load knots from map and re-leash mobs that were leashed before server close.
- Knots sound.
- Support from 1.8 to 1.12 protocol versions.

It's been a lot of work and testing but that was very enjoyable :P

Fixes #2915 and #2729

**Tests done:**
- Create some leads ( /i lead 24 ). They are stack-able now, as in vanilla.
- Go for passive mobs (pigs, cows, horses...) and right click them with a lead on main hand. They will be leashed to you. 
- Try on others than passive mobs (wont work).
- Take more than one mob at a time. You are now Santa Claus.
- Look them following you.
- Try to go far from the mob (> 10 blocks). Leads should break with no drop.
- Unleash a leashed mob right clicking on it, then used lead will become a pickup.
- Kill the mob while it's leashed to you, leash link (the drawn rope between you and the mob) will go out and there will be no drop.
- Catch a mob with a lead and then right click on a fence, a leash knot will show up. The leash sound should be heard. The mob will unleash from your player and get leashed to the knot.
- Try to leash more mobs to the same fence block. No more knots will be created, the existing will be reused. No sound will be reproduced.
- Try to leash more mobs to the knot (it's different to the server right clicking on a knot than right click on the fence sustaining the knot). It will work.
- Try to break a knot, all mobs leashed will be released.
- Try to break a fence where there is a knot. As in vanilla, the knot will break passed about a second, and mobs will be released.
- Try to leash a mob that's currently leashed to a fence leash knot. No effect and no rope line drawing. That last has been tricky to fix, because client anticipates to server and draws the line of the leash between player and the mob without regard from the current state of the mob.
- Try to put several leash knots on adjacent fence blocks, logout, close server, open server again, login. Every mob will be leashed to it's knot again.


**Not implemented:**
- Leashed mobs in vanilla can go around up to 5 blocks distance from the knot. Know mobs just go to the target pos like in other situations.
- Surely I forgot some LUA and doc stuff
